src/core/net/SRP_CODER.md

```diff
@@ -0,0 +1,309 @@
+- The remaining bits after the MSB provide the numerical `uint` bit values in big-endian order.
+
+### Examples
+
+- Numbers less than (2^7) 128 are encoded as a single byte.
+- Numbers between 128 and (2^14 - 1) 16,383 are encoded as two bytes.
+- Numbers between 16,384 up to (2^21 - 1) 2,097,151 are encoded as three bytes.
+
+## DNS Name and Label Format
+
+### Label Dispatch
+
+```
+    0   1   2   3   4   5   6   7
+  +---+---+---+---+---+---+---+---+
+  | Type  ... | ...               |
+  +---+---+---+---+---+---+---+---+
+
+```
+
+- `00 <6-bit len>`: Standard label encoding, where the lower 6 bits indicate the label's length (number of characters). This is followed by the label characters, totaling `len` in number.
+- `01 <6-bit len>`: Similar to the above, but the label begins with an underscore `'_'`. The `len` value here doesn't include the underscore itself, which isn't explicitly encoded after the dispatch byte.
+- `10 <6-bit offset first seg>`: This references a previously encoded label within the message. The 6-bit value in the dispatch byte serves as the first segment of a compact `uint` representing the offset to that label. This offset must point to the label dispatch byte of the previous label.
+  - Note: This mechanism applies to a single label only. Any subsequent labels are encoded directly after the current one. This differs from traditional DNS pointer name compression, where a pointer indicates that the remaining labels should be read from the specified offset.
+- `110 <5-bit code>`: Commonly used constant labels. Codes are:
+  - `0`: `_udp`
+  - `1`: `_tcp`
+  - `2`: `_matter`
+  - `3`: `_matterc`
+  - `4`: `_matterd`
+  - `5`: `_hap`
+- `111 <5-bit code>`: Commonly Used Generative Patterns
+  - `0`: `<hex_value>` - 16-character uppercase hexadecimal 64-bit value.
+    - Example: `DAAFF10F39B00F32`
+    - Encoding: After the label dispatch byte, the 64-bit value is encoded as 8-byte binary format (big-endian order).
+  - `1`: `<hex_value_1>-<hex_value_2>` - Two 16-character uppercase hexadecimals, separated by a hyphen `-`
+    - Example `AABBCCDDEEFF0011-1122334455667788`.
+    - Encoding: After the label dispatch byte, the two 64-bit values are encoded (each as 8-byte binary format big-endian).
+  - `2`: Subtype label `_<char><hex_value>` - starting with an underscore `_`, followed by a single character `<char>`, ending with a 16-character uppercase hexadecimal 64-bit value `<hex_value>`.
+    - Example: `_IAA557733CC00EE11`
+    - Encoding: After the label dispatch byte, `<char>` is encoded as a single byte, followed by the 8-byte big-endian binary representation of the 64-bit value `<hex_value>`.
+  - `3`: Subtype Label - same as the previous case.
+    - Encoding: Instead of directly encoding the 8-byte value, an offset pointing to an earlier occurrence of the same byte sequence within the message is encoded. The offset is encoded using a compact `uint` format.
+
```

Comment:
Is this also intended to point to the label dispatch byte?

Reply:
Good question. Updated the text to make it clear. The offset points to the first byte of an earlier occurrence of the same 8-byte sequence within the message.

Reply:
Oh, I think I was missing something here. This could be pointing back into a 16-byte sequence, couldn't it? I was assuming just to a previous _<char><sequence>, which is why I thought it could be pointing at the header byte. So it would be worth saying that explicitly, although I think you do already say it, just not explicitly enough. :)